From c7d492b32d190c2e937225cdbe05ca3f14d7335f Mon Sep 17 00:00:00 2001
From: Dan Kephart <dan.kephart@lairdtech.com>
Date: Tue, 3 Feb 2015 13:07:40 -0500
Subject: [PATCH] laird: ath6kl: Allow connection to AP with multiple ciphers

The WB45 was not connecting to an AP configured for WPA2 but allowing
both AES and TKIP encryption.  When the WB45 was configured for wpa2-aes,
it would not connect to this mixed mode AP configuration.  It should
connect using WPA2 with AES as the unicast cipher and TKIP for the group
cipher but the WB45 would not attempt to connect (probes sent and probe
responses, but never attempted to associate).

By or'ing in the ignore wpax cipher flag resolves the issue.

Bug: 5014
---
 drivers/net/wireless/ath/ath6kl/cfg80211.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath6kl/cfg80211.c b/drivers/net/wireless/ath/ath6kl/cfg80211.c
index ae0570911ee7..ff7933b0ac04 100644
--- a/drivers/net/wireless/ath/ath6kl/cfg80211.c
+++ b/drivers/net/wireless/ath/ath6kl/cfg80211.c
@@ -638,7 +638,8 @@ static int ath6kl_cfg80211_connect(struct wiphy *wiphy, struct net_device *dev,
 					vif->grp_crypto, vif->grp_crypto_len,
 					vif->ssid_len, vif->ssid,
 					vif->req_bssid, vif->ch_hint,
-					ar->connect_ctrl_flags, nw_subtype);
+					ar->connect_ctrl_flags | CONNECT_IGNORE_WPAx_GROUP_CIPHER, 
+					nw_subtype);
 
 	if (sme->bg_scan_period == 0) {
 		/* disable background scan if period is 0 */
-- 
2.11.0

